APE26.rst

```diff
@@ -0,0 +1,325 @@
+changes.
+
+Several issues have been raised regarding topics such as
+confusion differentiating the use of ``frame`` and ``SkyCoord``
+for data storage, and problems arising in other astropy
+subpackages when using frames that store data. For example:
+
+- `Comparing Frame with data and SkyCoord with same data raises 
+  exception #13476 
+  <https://github.com/astropy/astropy/issues/13476>`_
+- `Add Frame objects without data to a Table #16823 
+  <https://github.com/astropy/astropy/issues/16823>`_
+
+Additionally, multiple pull requests have factored out common 
+code between frames and ``SkyCoord``, showing that there is no 
+proper separation of concern:
+
+- `Allow BaseCoordinateFrames to be stored in tables (by giving 
+  them .info) #16831 <https://github.com/astropy/astropy/pull/16831>`_
+- `Masked frames and SkyCoord #17106 
+  <https://github.com/astropy/astropy/pull/17016>`_ (this was later 
+  removed and instead methods were duplicated)
+
+Further, pull requests have added methods to make frames and 
+``SkyCoord`` even more similar, underscoring that frames *with* 
+data should not be separate entities from ``SkyCoord``:
+
+- `Implement BaseCoordinateFrame.to_table() #17009 
+  <https://github.com/astropy/astropy/pull/17009>`_
+- `Implement BaseCoordinateFrame.frame property #16356 
+  <https://github.com/astropy/astropy/pull/16356>`_
+
+In addition, many of these ideas have been developed and tested in
+parallal in the JAX-oriented library `coordinax
+<https://github.com/GalacticDynamics/coordinax>`_. Many of the
+developers of that library are also active Astropy developers and
+the development effort towards ``coordinax`` informs, tests, and
+validates the ideas presented in this APE. In short, it works.
+
+
+Implementation
+--------------
+The direct use of coordinate frames instead of ``SkyCoord`` is 
+common. In particular ``ICRS`` objects are frequently created 
+with data. Given the prevalent use, it is imperative to maintain 
+backward compatibility and not break the API too quickly. 
+Therefore, we propose implementing this APE through 4 steps (and 
+substeps).
+
+1. Splitting the frame classes into two hierarchies: ones with 
+   and without data, with the data-less ones getting new names.
+
+2. Adding a new ``Coordinate`` class that is similar to 
+   ``SkyCoord``, but which does not keep any frame attributes not 
+   in the current frame, and does not have extra features like 
+   caching and flexible input parsing. It will only accept 
+   data-less frame classes.
+
+3. Switching ``SkyCoord`` to use the data-less frame classes, and 
+   enabling automatic conversion of the with-data frames into 
+   ``SkyCoord`` objects.
+
+4. Deprecating the legacy with-data frame classes.
+
+   - Emitting warnings when instantiated.
+
+   - Still warn, but return a ``Coordinate``, not an instance of 
+     its class type (by overriding ``__new__``)
+
+   - Remove.
+
+The third step (at stage 3a) is illustrated in the following 
```

Comment:
What exactly is `stage 3a` again? I don't see any `a.` sub-bullet until Step 3 above.

Reply:
Yeah, the first sub-bullet in Step 3. Except now it's called Step 4. Needs to be updated.

Reply:
Done in [a7c80f2](https://github.com/astropy/astropy-APEs/pull/112/commits/a7c80f2b3ce36297d0ee3904b718ef0cf71c34a8).